electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,317 @@ def validate_value_length(value: bytes):
+
+    except Exception as e:
+        raise ValueError(f"{e}")
+
+def validate_zeronet_address(address: str) -> None: # Validate P2PKH Address (Zero Net)
+    try:
+        addrtype, _ = b58_address_to_hash160(address)
+    except Exception as e:
+        raise ValueError(f"Invalid zeronet address: {e}")
+    if addrtype != ZERONET_BASE58_ADDRTYPE:
+        raise ValueError("Invalid address type.")
+
+def validate_relative_domain(value: str, /, *, rfc_1034: bool = False, rfc_2782: bool = False):
+        if not value:
+            return False
+        try:
+            return not re.search(r"\s", value) and re.match(
+                # First character of the domain
+                rf"^(?:[a-zA-Z0-9{'_'if rfc_2782 else ''}]"
+                # Sub domain + hostname
+                + r"(?:[a-zA-Z0-9-_]{0,61}[A-Za-z0-9])?\.)"
+                # First 61 characters of the gTLD
+                + r"+[A-Za-z0-9][A-Za-z0-9-_]{0,61}"
+                # Last character of the gTLD
+                + rf"[A-Za-z]{r'.$' if rfc_1034 else r'$'}",
+                value.encode("idna").decode("utf-8"),
+                re.IGNORECASE,
+            )
+        except UnicodeError:
+            return False
+
+def validate_domains(domain: str) -> None:
+    if domain.endswith('.'):
+        relative_domain = domain[:-1]
+        if not validate_relative_domain(relative_domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+    else:
+        if not validate_relative_domain(domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+        try:
+            # Read the TLD set from the JSON file
+            current_directory = os.path.dirname(os.path.abspath(__file__))
+            json_file_path = os.path.join(current_directory, 'tld_list.json')
+
+            with open(json_file_path, 'r') as f:
+                tld_set = set(json.load(f))
+        except Exception as e:
+            raise ValueError(f"Unable to read the TLD list: {e}")
+
+        TLD = domain.split('.')[-1].lower()
+        if tld_set and TLD.upper() in tld_set:
+            raise ValueError(f'Recognized TLD which might indicate an invalid domain: {domain}')
+
+def validate_nameservers(domain: str) -> None:
+    if domain.endswith('.'):
+        relative_domain = domain[:-1]
+        if not validate_relative_domain(relative_domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+    else:
+        if not validate_relative_domain(domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+        try:
+            # Read the TLD set from the JSON file
+            current_directory = os.path.dirname(os.path.abspath(__file__))
+            json_file_path = os.path.join(current_directory, 'tld_list.json')
+
+            with open(json_file_path, 'r') as f:
+                tld_set = set(json.load(f))
+        except Exception:
+            # If the JSON file is not found, we will skip the TLD validation
```

Comment:
If the JSON file is not found, I think that should trigger an error; there's no valid reason why that file would be missing, is there?